tools/PsesPsClient/PsesPsClient.psm1

```diff
@@ -0,0 +1,314 @@
+#
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+$script:PsesBundledModulesDir = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath(
+    "$PSScriptRoot/../../../../module")
+
+class PsesStartupOptions
+{
+    [string]   $LogPath
+    [string]   $LogLevel
+    [string]   $SessionDetailsPath
+    [string[]] $FeatureFlags
+    [string]   $HostName
+    [string]   $HostProfileId
+    [version]  $HostVersion
+    [string[]] $AdditionalModules
+    [string]   $BundledModulesPath
+    [bool]     $EnableConsoleRepl
+}
+
+class PsesServerInfo
+{
+    [pscustomobject]$SessionDetails
+
+    [System.Diagnostics.Process]$PsesProcess
+
+    [PsesStartupOptions]$StartupOptions
+}
+
+function Start-PsesServer
+{
+    [CmdletBinding(SupportsShouldProcess)]
+    [OutputType([PsesServerInfo])]
+    param(
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $EditorServicesPath = "$script:PsesBundledModulesDir/PowerShellEditorServices/Start-EditorServices.ps1",
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $LogPath,
+
+        [Parameter()]
+        [ValidateSet("Diagnostic", "Normal", "Verbose", "Error")]
+        [string]
+        $LogLevel = 'Diagnostic',
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $SessionDetailsPath,
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [string[]]
+        $FeatureFlags = @('PSReadLine'),
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $HostName = 'PSES Test Host',
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $HostProfileId = 'TestHost',
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [version]
+        $HostVersion = '1.99',
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [string[]]
+        $AdditionalModules = @('PowerShellEditorServices.VSCode'),
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $BundledModulesPath,
+
+        [Parameter()]
+        [switch]
+        $EnableConsoleRepl
+    )
+
+    $EditorServicesPath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($EditorServicesPath)
+
+    $instanceId = Get-RandomHexString
+
+    $tempDir = [System.IO.Path]::GetTempPath()
+
+    if (-not $LogPath)
+    {
+        $LogPath = Join-Path $tempDir "pseslogs_$instanceId.log"
+    }
+
+    if (-not $SessionDetailsPath)
+    {
+        $SessionDetailsPath = Join-Path $tempDir "psessession_$instanceId.log"
+    }
+
+    if (-not $BundledModulesPath)
+    {
+        $BundledModulesPath = $script:PsesBundledModulesDir
+    }
+
+    $editorServicesOptions = @{
```

Comment:
I don't really like the idea of the array, mainly because it's heterogenous and the parameters need `-` before them.

I've already written the hashtable conversion logic, so it's done now.

What I really want is [`ConvertTo-Psd`](https://github.com/nightroman/PsdKit)